src/history/hs_rec.c

```diff
@@ -271,7 +272,10 @@ __hs_next_upd_full_value(WT_SESSION_IMPL *session, WT_UPDATE_VECTOR *updates,
         full_value->data = upd->data;
         full_value->size = upd->size;
     } else if (upd->type == WT_UPDATE_MODIFY) {
-        WT_RET(__wt_buf_set(session, full_value, older_full_value->data, older_full_value->size));
+        max_memsize = older_full_value->size;
+        __wt_modify_max_memsize_format(upd->data, S2BT(session)->value_format, &max_memsize);
+        WT_RET(__wt_buf_set_and_grow(
```

Comment:
why can't we use the `__wt_buf_set` function here by passing the max value fo the current size and expected size?

Reply:
buf set function only grows the data to the size of the copy. We need to grow the buffer to larger than that to cater the future growth of the buffer when applying the modifies.

Reply:
If we pass the size as `WT_MAX( buf_size, size)` as an argument, it should do the same job as you expected.

Reply:
I think if we do that, it will copy more data then required.

Reply:
Ok, understood. We need to set the current buf size to the value size, but increase the buffer to the maximum possible value size.